drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,482 @@
+ * @brief Update selected I2C register contents
+ * @param dev   	  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param update_val  - Value used to update on selected register
+ * @param update_mask - Mask to update to selected register bit/s
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_reg_update(struct max9611_dev *dev, uint8_t addr,
+		       int update_val, int update_mask)
+{
+	uint8_t temp;
+	int ret;
+
+	ret = max9611_read(dev, addr, 1, &temp);
+	if (ret)
+		return ret;
+
+	temp &= ~update_mask;
+	temp |= update_val;
+
+	return max9611_write(dev, addr, 1, &temp);
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 operating mode
+ * @param dev 		  - The device structure.
+ * @param mode 		  - MODE configuration
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_mode(struct max9611_dev *dev, enum max9611_mode_conf mode)
+{
+	mode = no_os_field_prep(MAX9611_MODE_MASK, mode);
+
+	return max9611_reg_update(dev, MAX9611_CTR1, mode,
+				  MAX9611_MODE_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get the MAX9611 operating mode
+ * @param dev 		  - The device structure.
+ * @param mode 		  - Pointer to store the obtained mode
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_mode(struct max9611_dev *dev, enum max9611_mode_conf *mode)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*mode = no_os_field_get(MAX9611_MODE_MASK, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures the internal MUX settings
+ * @param dev 		  - The device structure.
+ * @param mux 		  - MUX configuration
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_mux(struct max9611_dev *dev, enum max9611_mux_conf mux)
+{
+	switch (mux) {
+	case MAX9611_CONF_SENSE_1X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_SENSE_4X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_SENSE_8X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_IN_COM_MODE:
+		dev->capture_type = MAX9611_DATA_RSP;
+		break;
+	case MAX9611_CONF_OUT_ADC:
+		dev->capture_type = MAX9611_DATA_OUT;
+		break;
+	case MAX9611_CONF_SET_ADC:
+		dev->capture_type = MAX9611_DATA_SET;
+		break;
+	case MAX9611_CONF_TEMP:
+		dev->capture_type = MAX9611_DATA_TMP;
+		break;
+	default:
```

Comment:
ditto: we should cover also the case where the user passes a wrong value as input and return an error code.

Reply:
Fast mode configuration added in case to default to 1x Current Sense amplifier setting and return -EINVAL for invalid inputs as error code in latest commit.